datalad_next/create_sibling_webdav.py

```diff
@@ -0,0 +1,157 @@
+from datalad.support.param import Parameter
+from datalad.support.annexrepo import AnnexRepo
+from datalad.support.constraints import (
+    EnsureNone,
+    EnsureStr,
+)
+
+
+__docformat__ = "restructuredtext"
+
+lgr = logging.getLogger('datalad_next.create_sibling_webdav')
+
+
+class CreateSiblingWebDAV(Interface):
+    """
+
+    """
+    _examples_ = [
+    ]
+
+    _params_ = dict(
+        url=Parameter(
+            args=("url",),
+            metavar="https://<host>[:<port>]/<local part>",
+            doc="URL identifying the target WebDAV server",
+            constraints=EnsureStr()),
+        dataset=Parameter(
+            args=("-d", "--dataset"),
+            doc="""specify the dataset to process.  If
+            no dataset is given, an attempt is made to identify the dataset
+            based on the current working directory""",
+            constraints=EnsureDataset() | EnsureNone()),
+        name=Parameter(
+            args=('-s', '--name',),
+            metavar='NAME',
+            doc="Name of the sibling.",
+            constraints=EnsureStr() | EnsureNone(),
+            required=True),
+        credential=Parameter(
+            args=("--credential",),
+            doc="""Name of the credentials that should be used to access
+            the WebDAV server.""",
+        )
+    )
+
+    @staticmethod
+    @datasetmethod(name='create_sibling_webdav')
+    @eval_results
+    def __call__(
+            url: str,
+            *,
+            dataset: Optional[Union[str, Dataset]] = None,
+            name: Optional[str] = None,
+            credential: Optional[str] = None):
+        """
+
+        :param url:
+        :param dataset:
+        :param name:
+        :param credential:
+        :return: a generator yielding result records
+        """
+
+        ds = require_dataset(
+            dataset or ".",
+            check_installed=True,
+            purpose=f'create WebDAV sibling(s)')
+
+        res_kwargs = dict(
+            action=f'create_sibling_webdav',
+            logger=lgr,
+            refds=ds.path,
+        )
+
+        if not isinstance(ds.repo, AnnexRepo):
+            yield {
+                **res_kwargs,
+                "status": "error",
+                "msg": "require annex repo to create WebDAV sibling"
+            }
+            return
+
+        parser_url = urlparse(url)
+        if parser_url.query:
+            yield {
+                **res_kwargs,
+                "status": "error",
+                "msg": "URLs with query are not yet supported"
+            }
+            return
+
+        name_base = name or parser_url.hostname
+        git_name = name_base + "-wd-vcs"
+        annex_name = name_base + "-wd-tree"
+
+        datalad_annex_url = (
+            "datalad_annex::"
+            + url
+            + "?type=webdav&url={noquery}&encryption=none&exporttree=yes"
+            + "" if credential is None else f"&dlacredential={credential}"
```

Comment:
This raises an interesting case. It is technically not necessary to do `datalad-annex:: + URL + ?type=webdav&url={noquery}&....` it could just as well be `datalad-annex::?type=webdav&url={url}...`

But I do not think that the `datalad-annex` implementation actually tests URLs with a query component. I strongly suspect that it is possible with the right kind of quoting, but a test plus the resulting documentation is needed.

Reply:
I also think it must be `datalad-annex::` not `datalad_annex::`

Reply:
done